src/view/dataElements/xdmObject.jsx

```diff
@@ -164,38 +285,87 @@ const XdmExtensionView = () => {
       }}
       getSettings={({ values }) => {
         return {
+          sandbox: {
+            name: selectedSandboxMeta ? selectedSandboxMeta.name : null
+          },
           schema: {
-            id: selectedSchemaMeta.$id,
-            version: selectedSchemaMeta.version
+            id:
+              selectedSchemaMeta && selectedSchemaMeta.$id
+                ? selectedSchemaMeta.$id
+                : null,
```

Comment:
@garyr, do you mind describing when we get into `getSettings` and have a null `selectedSchemaMeta`? That seems like what this check is for anyway. It seems like we should make validation fail if the user hasn't selected a schema, right?

When the extension is running inside Launch, if the user hits Save, Launch calls `validate` on the extension view. If `validate` returns `false`, then Launch won't call `getSettings`. Launch only calls `getSettings` if `validate` returns `true`. So, in other words, if we make `validate` return false if no schema has been selected, then we shouldn't have to do this null check here in `getSettings`. Let me know if that doesn't make sense.

Reply:
Okay, I will look at addressing this